lib/persisted-model.js

```diff
@@ -1041,6 +1042,13 @@ PersistedModel.enableChangeTracking = function() {
   assert(this.dataSource, 'Cannot enableChangeTracking(): ' + this.modelName +
     ' is not attached to a dataSource');
 
+  var idName = this.getIdName();
+  var idProp = this.definition.properties[idName];
+  if (!idProp || idProp.type !== String || !/^[ug]uid$/.test(idProp.defaultFn)) {
+    deprecated('The model ' + this.modelName + ' is tracking changes, ' +
+      'which requries a string id with GUID/UUID default value.');
+  }
+
   Change.attachTo(this.dataSource);
```

Comment:
Shall I rewrite it as `!(id.Prop.defaultFn === 'uuid' || id.Prop.defaultFn === 'guid')`?